slices/filter_map.go

```diff
@@ -0,0 +1,49 @@
+package slices
+
+import "github.com/akitasoftware/go-utils/optionals"
+
+// Applies f to each element of slice in order, removes any None results, and
+// returns the rest.
+func FilterMap[T1, T2 any](slice []T1, f func(T1) optionals.Optional[T2]) []T2 {
+	result, _ := FilterMapIndexWithErr(slice, func(_ int, t T1) (optionals.Optional[T2], error) {
+		return f(t), nil
+	})
+	return result
+}
+
+// Applies f to each element of slice in order, removes any None results, and
+// returns the rest. Returns an error if f returns a non-nil error on any
```

Comment:
Would it be worth adding that iteration stops immediately upon any error?

Reply:
Amended in c8cc8a8.